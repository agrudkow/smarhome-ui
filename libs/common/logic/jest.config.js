module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
