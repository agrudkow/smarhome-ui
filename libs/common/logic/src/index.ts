export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
