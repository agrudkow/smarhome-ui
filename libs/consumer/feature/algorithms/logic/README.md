# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
