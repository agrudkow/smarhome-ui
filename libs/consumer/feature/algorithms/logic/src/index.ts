export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
