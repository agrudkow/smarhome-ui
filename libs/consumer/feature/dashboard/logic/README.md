# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
