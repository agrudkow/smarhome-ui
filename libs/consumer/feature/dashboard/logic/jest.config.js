module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
