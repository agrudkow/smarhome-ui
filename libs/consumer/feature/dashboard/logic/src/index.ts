export * from './lib/use-billings';
