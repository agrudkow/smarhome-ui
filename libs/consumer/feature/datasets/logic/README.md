# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
