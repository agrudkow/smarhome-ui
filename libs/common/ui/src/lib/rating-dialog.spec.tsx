import React from 'react';
import { customRender } from '@smarthome/common/logic';
import RatingDialog from './rating-dialog';
import { fireEvent } from '@testing-library/react';

describe(' RatingDialog', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <RatingDialog
        open={true}
        title=""
        description=""
        rating={0}
        onClose={jest.fn()}
        onRatingChange={jest.fn()}
        onSend={jest.fn()}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <RatingDialog
        open={true}
        title=""
        description=""
        rating={0}
        onClose={jest.fn()}
        onRatingChange={jest.fn()}
        onSend={jest.fn()}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should call handler function on send rating', () => {
    const handleSend = jest.fn();
    const { getByText } = customRender(
      <RatingDialog
        open={true}
        title=""
        description=""
        rating={4}
        onClose={jest.fn()}
        onRatingChange={jest.fn()}
        onSend={handleSend}
      />
    );
    fireEvent.click(getByText('Send'));
    expect(handleSend).toHaveBeenCalledTimes(1);
  });

  it('should call handler function on close rating dailog', () => {
    const handleClose = jest.fn();
    const { getByText } = customRender(
      <RatingDialog
        open={true}
        title=""
        description=""
        rating={4}
        onSend={jest.fn()}
        onRatingChange={jest.fn()}
        onClose={handleClose}
      />
    );
    fireEvent.click(getByText('Close'));
    expect(handleClose).toHaveBeenCalledTimes(1);
  });
});
