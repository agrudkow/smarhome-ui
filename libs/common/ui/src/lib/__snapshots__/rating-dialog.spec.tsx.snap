// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` RatingDialog should match snapshot 1`] = `
.c2 {
  color: #555;
  border-color: #555;
  -webkit-transition: color 0.3s ease-in-out,border-color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out,border-color 0.3s ease-in-out;
  font-weight: 550;
  width: 100%;
}

.c2:hover {
  color: #1BBDCE;
  border-color: #1BBDCE;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

@media (max-width:480px) {
  .c1 {
    padding: 0;
  }
}

<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="rating-dialog"
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root"
          id="'rating-dialogg-title"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          />
        </div>
        <div
          class="MuiDialogContent-root"
        >
          <p
            class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          />
          <div
            class="c0"
          >
            <span
              class="MuiRating-root MuiRating-sizeLarge"
            >
              <span
                class="MuiRating-decimal"
              >
                <label
                  class="MuiRating-label"
                  for="rating-dialog-0-5"
                  style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    0.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-0-5"
                  name="rating-dialog"
                  type="radio"
                  value="0.5"
                />
                <label
                  class="MuiRating-label"
                  for="rating-dialog-1"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    1 Star
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-1"
                  name="rating-dialog"
                  type="radio"
                  value="1"
                />
              </span>
              <span
                class="MuiRating-decimal"
              >
                <label
                  class="MuiRating-label"
                  for="rating-dialog-1-5"
                  style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    1.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-1-5"
                  name="rating-dialog"
                  type="radio"
                  value="1.5"
                />
                <label
                  class="MuiRating-label"
                  for="rating-dialog-2"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    2 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-2"
                  name="rating-dialog"
                  type="radio"
                  value="2"
                />
              </span>
              <span
                class="MuiRating-decimal"
              >
                <label
                  class="MuiRating-label"
                  for="rating-dialog-2-5"
                  style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    2.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-2-5"
                  name="rating-dialog"
                  type="radio"
                  value="2.5"
                />
                <label
                  class="MuiRating-label"
                  for="rating-dialog-3"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    3 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-3"
                  name="rating-dialog"
                  type="radio"
                  value="3"
                />
              </span>
              <span
                class="MuiRating-decimal"
              >
                <label
                  class="MuiRating-label"
                  for="rating-dialog-3-5"
                  style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    3.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-3-5"
                  name="rating-dialog"
                  type="radio"
                  value="3.5"
                />
                <label
                  class="MuiRating-label"
                  for="rating-dialog-4"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    4 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-4"
                  name="rating-dialog"
                  type="radio"
                  value="4"
                />
              </span>
              <span
                class="MuiRating-decimal"
              >
                <label
                  class="MuiRating-label"
                  for="rating-dialog-4-5"
                  style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    4.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-4-5"
                  name="rating-dialog"
                  type="radio"
                  value="4.5"
                />
                <label
                  class="MuiRating-label"
                  for="rating-dialog-5"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyhidden"
                  >
                    5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyhidden"
                  id="rating-dialog-5"
                  name="rating-dialog"
                  type="radio"
                  value="5"
                />
              </span>
            </span>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing"
        >
          <div
            class="c1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined c2 MuiButton-outlinedPrimary"
              style="margin: 5px 0px 0px;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Send
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined c2 MuiButton-outlinedPrimary"
              style="margin: 5px 0px 0px;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Close
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
